src/encryption/encryption.go

```diff
@@ -103,3 +106,60 @@ func LogGroups(groups []Group, logger *zap.SugaredLogger) {
 		}
 	}
 }
+
+func (v *Value) encryptedPath() string {
+	return "#" + v.path.String()
+}
+
+func prepareMapToEncrypt(values []Value) map[string]string {
+	result := make(map[string]string)
+	for _, value := range values {
+		result[value.encryptedPath()] = value.value
+	}
+	return result
+}
+
+func DoEncrypt(projectState *state.State, unencryptedGroups []Group, api *Api) error {
+	errors := utils.NewMultiError()
+	localManager := projectState.LocalManager()
+	projectId := fmt.Sprintf("%v", projectState.Manifest().Project.Id)
+	for _, group := range unencryptedGroups {
+		var encryptionError error
+		// create map with {"#<value-path>":"<value-to-encrypt>"...} entries
+		mapToEncrypt := prepareMapToEncrypt(group.values)
+
+		//type switch on config or configRow state
+		switch o := group.object.(type) {
+		case *model.ConfigState:
+			encryptedMap, encryptionError := api.EncryptMapValues(o.ComponentId, projectId, mapToEncrypt)
+			if encryptionError != nil {
+				errors.Append(encryptionError)
+			}
+			//update local state with encrypted values
+			for _, value := range group.values {
+				encryptedValue := encryptedMap[value.encryptedPath()]
+				o.Local.Content = utils.UpdateIn(o.Local.Content, value.path, encryptedValue)
+			}
+		case *model.ConfigRowState:
+			encryptedMap, encryptionError := api.EncryptMapValues(o.ComponentId, projectId, mapToEncrypt)
+			if encryptionError != nil {
+				errors.Append(encryptionError)
+			}
+			//update local state with encrypted values
+			for _, value := range group.values {
+				encryptedValue := encryptedMap[value.encryptedPath()]
+				o.Local.Content = utils.UpdateIn(o.Local.Content, value.path, encryptedValue)
+			}
+		}
+		// skip saving config to disk if errors encountered during encryption or previous iterations
+		if errors.Len() > 0 || encryptionError != nil {
```

Comment:
To je trocha nekonzistentne, ... predchadzajuce configs sa zapisali na disk ... teraz nastane chyba ... a nasledne sa uz nezapisu ... ale tie prve zostanu zasifrovane.

Zapisoval by som to na disk vzdy pokial nenastala chyba pre tento config, ... ved cielom je vsetko zasifrovat.

Reply:
ok dal som ze sa config zapise na disk len ak sa nepodari encrypt api call pre dany config